Source/DafnyCore/Compilers/Compiler-java.cs

```diff
@@ -2217,14 +2217,12 @@ private class GenericArrayElementLvalue : ILvalue {
     private void EmitRuntimeJar(string targetDirectory) {
       // Since DafnyRuntime.jar is binary, we can't use ReadRuntimeSystem
       var jarName = "DafnyRuntime.jar";
-      var assembly = System.Reflection.Assembly.GetExecutingAssembly();
+      var assembly = System.Reflection.Assembly.Load("DafnyPipeline");
       var stream = assembly.GetManifestResourceStream(jarName);
-      if (stream is not null) {
-        var fullJarName = $"{targetDirectory}/{jarName}";
-        FileStream outStream = new FileStream(fullJarName, FileMode.Create, FileAccess.Write);
-        stream.CopyTo(outStream);
-        outStream.Close();
-      }
+      var fullJarName = $"{targetDirectory}/{jarName}";
+      FileStream outStream = new FileStream(fullJarName, FileMode.Create, FileAccess.Write);
+      stream!.CopyTo(outStream);
```

Comment:
This change has the side effect of creating an empty `JavaRuntime.jar` if it's not found in the assembly, as opposed to doing nothing, which is perhaps better behavior? It might be even nicer to emit an error message in that case, but I didn't do that in the original code, either. :)

Reply:
Ah my intention was just to cleanly make this an exception rather than a silent failure, but I can throw right away without that messy side effect.

Note that I'm going to make it an exception rather than an error message as `SinglePassCompiler.ReadRuntimeSystem` does (and I intend to make that throw as well in the next PR). At least right now, the publicly released tools should always have the runtimes attached, so I'd rather crash loudly if we've made a publishing mistake.

Reply:
In fact I'm touching `SinglePassCompiler.ReadRuntimeSystem` already anyway, I'm going to make it throw in this PR after all - no time like the present!